app/services/mapping/declaration/AgentBuilder.scala

```diff
@@ -28,47 +28,51 @@ object AgentBuilder {
   def build(implicit cacheMap: CacheMap): Declaration.Agent =
     cacheMap
       .getEntry[RepresentativeDetails](RepresentativeDetails.formId)
-      .map(data => createAgent(data.details.orNull))
+      .filter(isDefined)
+      .map(data => createAgent(data.details.get))
       .orNull
 
+  private def isDefined(representativeDetails: RepresentativeDetails): Boolean =
+    representativeDetails.details.isDefined && (representativeDetails.details.get.eori.isDefined || representativeDetails.details.get.address.isDefined)
+
   private def createAgent(details: EntityDetails): Declaration.Agent = {
     val agent = new Declaration.Agent()
 
-    val agentId = new AgentIdentificationIDType()
-    agentId.setValue(details.eori.orNull)
-    agent.setID(agentId)
+    if (details.eori.isDefined) {
+      val agentId = new AgentIdentificationIDType()
+      agentId.setValue(details.eori.get)
+      agent.setID(agentId)
+    } else {
 
-    val agentAddress = new Agent.Address()
+      val agentAddress = new Agent.Address()
 
-    details.address.map(address => {
+      details.address.map(address => {
```

Comment:
Address is used in different mappings for all the pages , it would be good if we have a common method of extracting and mapping header if the the types are same.

Reply:
The return type is always of a different type.

Reply:
We could convert to a Map or Json and then create an instance of another type.
Let me that onboard and unblock our pipeline